lib/puppet/application/apply.rb

```diff
@@ -353,6 +353,10 @@ def read_catalog(text)
 
   def apply_catalog(catalog)
     configurer = Puppet::Configurer.new
-    configurer.run(:catalog => catalog, :pluginsync => false)
+    if Puppet[:facts_terminus] ==  :network_device
+      configurer.run(:network_device => true, :catalog => catalog, :pluginsync => false)
```

Comment:
Rather than having the apply application know this information, I'd prefer for the caller to pass options through, e.g. `apply.run_command({:network_device => true})`.

Reply:
I dont follow 
[1] pry(#<Puppet::Application::Device>)> apply.run_command({:network_device => true})
ArgumentError: wrong number of arguments (given 1, expected 0)
from /Users/tp/workspace/git/puppet/lib/puppet/application/apply.rb:170:in `run_command'
[2] pry(#<Puppet::Application::Device>)>

Reply:
You'll need to thread the options through to the configurer. I think it's preferable to take a set of options instead of having the apply application know anything about running in "device mode".

Reply:
correction 3 and a a bit weeks ago.